app/controllers/secrets_controller.rb

```diff
@@ -54,6 +54,43 @@ def duplicates
       sort_by { |_, v| -v.size }
   end
 
+  def resolve
+    if params[:project_id].present?
+      project = if params[:project_id].match?(/^\d+$/)
+        Project.find_by_id(params[:project_id])
+      else
+        Project.find_by_permalink(params[:project_id])
+      end
+    elsif params[:project_permalink].present?
+      project = Project.find_by_permalink(params[:project_permalink])
+    else
+      raise "Neither project_id or project_permalink given as parameters"
+    end
+
+    deploy_group = DeployGroup.find_by_permalink(params.require(:deploy_group))
+    keys_param = params.require(:keys)
+    keys = keys_param.is_a?(Array) ? keys_param : keys_param.split(/, ?/)
```

Comment:
support for 1 flavor should be enough ... arrays seem fine 🤷‍♂

Reply:
this was so you could easily resolve a couple of keys in the browser by doing `keys=key1,key2,key3` .. but i can remove it if we just want to support `keys[]=key1&keys[]=key2&.....`